### ISSUE TITLE
Viewer buttons from multiple viewers connected to last created viewer

### ISSUE THREAD

**Issue by gatoniel** [napari/napari#3764]
## 🐛 Bug

When using multiple viewers, the viewer buttons in the bottom left do not change the camera scene of the viewer they belong to. Instead, they change the camera scene of the last created viewer. This might be related to #1530 .

## To Reproduce

1. Run the following code to open two napari viewers
```
from napari import Viewer
from skimage.data import cells3d, astronaut


viewer1 = Viewer(title="viewer 1")
viewer1.add_image(cells3d()[30, 1])

viewer2 = Viewer(title="viewer 2")
viewer2.add_image(astronaut())
```

2. Go to `viewer 1` and click one of `Toggle ndisplay`, `Change order ...`, `Transpose order ...`.
3. The scene in `viewer 1` will not change.
4. Observe that the scene in `viewer 2` changes.
5. Close `viewer 2`.
6. Click on `Show/Hide IPython ...` in `viewer 1`. This results in the following error message
```
raceback (most recent call last):
  File "d:\napari-3d-ortho-viewer\.win-napari-3d-ortho-viewer\lib\site-packages\napari\utils\action_manager.py", line 53, in <lambda>
    self._command_with_context = lambda: call_with_context(
  File "d:\napari-3d-ortho-viewer\.win-napari-3d-ortho-viewer\lib\site-packages\napari\utils\action_manager.py", line 42, in call_with_context
    return function(**ctx)
  File "d:\napari-3d-ortho-viewer\.win-napari-3d-ortho-viewer\lib\site-packages\napari\_qt\qt_viewer.py", line 739, in toggle_console_visibility
    viz = not self.dockConsole.isVisible()
RuntimeError: wrapped C/C++ object of type QtViewerDockWidget has been deleted
```
7. Although `viewer 2` is closed now, clicking `Toggle ndisplay`, `Change order ...`, `Transpose order ...` in `viewer 1` does not have any effect on the scene of `viewer 1`.

## Expected behavior

I expect that clicking on the viewer buttons in `viewer 2` changes the scene in `viewer 2`. Clicking the buttons in `viewer 1` should change the scene in `viewer 1`.

## Environment

napari: 0.4.12
Platform: Windows-10-10.0.22000-SP0
Python: 3.9.5 (tags/v3.9.5:0a7dcbd, May 3 2021, 17:27:52) [MSC v.1928 64 bit (AMD64)]
Qt: 5.15.2
PyQt5: 5.15.6
NumPy: 1.21.4
SciPy: 1.7.3
Dask: 2021.11.2
VisPy: 0.9.4

OpenGL:
- GL version: 4.6.0 - Build 27.20.100.9664
- MAX_TEXTURE_SIZE: 16384

Screens:
- screen 1: resolution 2560x1440, scale 1.0
- screen 2: resolution 2560x1440, scale 1.0

Plugins:
- console: 0.0.4
- napari-3d-ortho-viewer: 0.0.1
- scikit-image: 0.4.12
- svg: 0.1.5

**Comment by tlambert03** [comment]
Hi @gatoniel, I believe this is the same thing that was just fixed in #3692 (i.e. duplicate of #2977 and #2897).  If you are able to work on the master branch, can you confirm?  (it would be `pip install -U https://github.com/napari/napari.git`) ... if not, you can wait until the next release to give it a try.  should be soon

**Comment by gatoniel** [comment]
Yes, works with current master branch.
